src/sidebar/store/modules/annotations.js

```diff
@@ -320,47 +387,48 @@ function updateAnchorStatus(statusUpdates) {
 }
 
 /**
- * Update the local hidden state of an annotation.
+ * Updating the flagged status of an annotation.
+ *
+ * @param {string} id - Annotation ID
+ * @param {boolean} isFlagged - The flagged status of the annotation. True if
+ *        the user has flagged the annotation.
  *
- * This updates an annotation to reflect the fact that it has been hidden from
- * non-moderators.
  */
-function hideAnnotation(id) {
+function updateFlagStatus(id, isFlagged) {
   return {
-    type: actions.HIDE_ANNOTATION,
+    type: actions.UPDATE_FLAG_STATUS,
     id: id,
+    isFlagged: isFlagged,
   };
 }
 
+/* Selectors */
+
 /**
- * Update the local hidden state of an annotation.
+ * Count the number of annotations (as opposed to notes or orphans)
  *
- * This updates an annotation to reflect the fact that it has been made visible
- * to non-moderators.
+ * @return {number}
```

Comment:
I don't think `@return` above a selector created with `const aSelector = createSelector(...)` works as expected. My guess is that `@return` only works with function or method declarations rather than variables. You won't get any error if you specify the wrong return type for example. Using `@type` works though:

```js
/**
 * Count the number of orphans currently in the collection
 *
 * @type {(state: any) => number}
 */
const orphanCount = createSelector(
  state => state.annotations.annotations,
  annotations => countIf(annotations, metadata.isOrphan)
);
```

eg. If you change `number` to `boolean` above, you'll get an error.

Reply:
Presumably this applies to all `createSelector` uses?

Reply:
Clarification: here, `@type` is asserting a type on the function itself?